File: pipeline/adapters/astropy_adapter/experiments/kepler.py
import json
import sys

try:
    from .adapter_perf import astropy_perf_skipped_result
except ImportError:
    from experiments.adapter_perf import astropy_perf_skipped_result


def run_kepler_solver(lines_iter):
    """Removed — Astropy has no public Kepler-equation solver.

    Reading the experiment input still consumes the expected number of
    lines so the orchestrator's stdin protocol stays in sync, but the
    result JSON declares the experiment unsupported.  The capability
    catalog also drops this row from the public scoreboard.
    """
    n = int(next(lines_iter).strip())
    for _ in range(n):
        next(lines_iter)
    result = {
        "experiment": "kepler_solver",
        "library": "astropy",
        "model": "unsupported",
        "status": "unsupported",
        "reason": (
            "Astropy does not expose a public Kepler-equation solver; "
            "row is excluded by the capability catalog."
        ),
        "count": 0,
        "cases": [],
    }
    json.dump(result, sys.stdout, indent=None)
    print()

def _legacy_run_kepler_solver_DELETED(lines_iter):
    """Removed Newton-Raphson solver (audit: fabricated upstream API).

    Body intentionally elided — see :func:`run_kepler_solver` for the
    catalog-aligned ``unsupported`` response.
    """
    raise RuntimeError("legacy implementation removed; see run_kepler_solver")

def run_kepler_solver_perf(lines_iter):
    """Removed perf workload — Astropy publishes no Kepler solver."""
    n = int(next(lines_iter).strip())
    for _ in range(n):
        next(lines_iter)
    astropy_perf_skipped_result(
        "kepler_solver_perf",
        "Astropy does not expose a public Kepler-equation solver.",
    )

File: pipeline/adapters/erfa_adapter/src/kepler.c
#include "common.h"

/* ------------------------------------------------------------------ */
/* Experiment: kepler_solver                                           */
/* Kepler equation M→E→ν self-consistency                              */
/* Input per line: M_rad  e                                            */
/* ------------------------------------------------------------------ */

void run_kepler_solver(void) {
    /* Removed: ERFA exposes no public Kepler-equation solver.  Catalog
     * marks (erfa, kepler_solver) as support="unsupported" so this
     * experiment never reaches the public scoreboard.  We still drain
     * stdin so the orchestrator's stream protocol stays in sync. */
    int n;
    if (scanf("%d", &n) != 1) { fprintf(stderr, "bad N\n"); exit(1); }
    for (int i = 0; i < n; i++) {
        double M_rad, e;
        if (scanf("%lf %lf", &M_rad, &e) != 2) {
            fprintf(stderr, "bad input line %d\n", i); exit(1);
        }
    }
    printf("{\"experiment\":\"kepler_solver\",\"library\":\"erfa\",");
    printf("\"model\":\"unsupported\",\"status\":\"unsupported\",");
    printf("\"reason\":\"ERFA does not expose a public Kepler-equation solver; row excluded by capability catalog.\",");
    printf("\"count\":0,\"cases\":[]}\n");
}
void run_kepler_solver_perf(void) {
    /* Removed: see run_kepler_solver above. */
    int n;
    if (scanf("%d", &n) != 1) { fprintf(stderr, "bad N\n"); exit(1); }
    for (int i = 0; i < n; i++) {
        double M_rad, e;
        if (scanf("%lf %lf", &M_rad, &e) != 2) {
            fprintf(stderr, "bad input line %d\n", i); exit(1);
        }
    }
    printf("{\"experiment\":\"kepler_solver_perf\",\"library\":\"erfa\",");
    printf("\"status\":\"unsupported\",\"reason\":\"ERFA does not expose a public Kepler-equation solver.\",");
    printf("\"count\":0,\"total_ns\":0,\"per_op_ns\":0,\"throughput_ops_s\":0,\"_sink\":0.0}\n");
}
